src/Apps/Settings/Routes/SavedSearchAlerts/Components/NewSavedSearchAlertEditForm.tsx

```diff
@@ -0,0 +1,419 @@
+import { Modal } from "Components/Alert/Components/Modal/Modal"
+import { DEFAULT_FREQUENCY } from "Components/SavedSearchAlert/constants"
+
+interface NewSavedSearchAlertEditFormQueryRendererProps {
+  editAlertEntity: EditAlertEntity
+  onDeleteClick: () => void
+  onCompleted: () => void
+  onCloseClick: () => void
+}
+
+interface NewSavedSearchAlertEditStepsProps {
+  savedSearch: NonNullable<
+    NewSavedSearchAlertEditFormQuery["response"]["me"]
+  >["savedSearch"]
+  viewer: NewSavedSearchAlertEditForm_viewer$data
+  onDeleteClick: () => void
+  onCompleted: () => void
+  onCloseClick: () => void
+}
+
+interface NewSavedSearchAlertEditFormProps {
+  viewer: NewSavedSearchAlertEditForm_viewer$data
+  onDeleteClick: () => void
+  onCompleted: () => void
+}
+const NewSavedSearchAlertEditSteps: React.FC<NewSavedSearchAlertEditStepsProps> = ({
+  viewer,
+  onDeleteClick,
+  onCompleted,
+  onCloseClick,
+}) => {
+  const { current } = useAlertContext()
+
+  return (
+    <>
+      <Media greaterThanOrEqual="md">
+        {current === "ALERT_DETAILS" && (
+          <Box flex={1} p={4}>
+            <Flex justifyContent="space-between">
+              <ModalHeader />
+              <Clickable onClick={onCloseClick}>
+                <CloseIcon display="flex" />
+              </Clickable>
+            </Flex>
+            <Spacer y={4} />
+            <NewSavedSearchAlertEditForm
+              viewer={viewer}
+              onDeleteClick={onDeleteClick}
+              onCompleted={onCompleted}
+            />
+          </Box>
+        )}
+        {current === "ALERT_FILTERS" && (
+          <Box flex={1} px={2} pt={4}>
+            <ModalHeader />
+            <Filters />
+            <FiltersFooter />
+          </Box>
+        )}
+      </Media>
+
+      <Media lessThan="md">
+        <Modal onClose={onCloseClick}>
+          <Flex mx={2}>
+            {current === "ALERT_DETAILS" && (
+              <NewSavedSearchAlertEditForm
+                viewer={viewer}
+                onDeleteClick={onDeleteClick}
+                onCompleted={onCompleted}
+              />
+            )}
+            {current === "ALERT_FILTERS" && <Filters />}
+          </Flex>
+        </Modal>
+      </Media>
+    </>
+  )
+}
+const NewSavedSearchAlertEditForm: React.FC<NewSavedSearchAlertEditFormProps> = ({
+  viewer,
+  onDeleteClick,
+  onCompleted,
+}) => {
+  const { state, goToFilters, dispatch, onCompleteEdit } = useAlertContext()
+
+  const isCustomAlertsNotificationsEnabled = viewer.notificationPreferences.some(
+    preference => {
+      return (
+        preference.channel === "email" &&
+        preference.name === "custom_alerts" &&
+        preference.status === "SUBSCRIBED"
+      )
+    }
+  )
+
+  return (
+    <Formik<AlertFormikValues>
+      initialValues={state.settings}
+      onSubmit={onCompleteEdit}
+      validateOnChange
+    >
+      {({ isSubmitting, values, setFieldValue, handleSubmit, dirty }) => {
```

Comment:
We need to check the form values, the array of pill values and notification settings (at least one has to be selected). In combination with `dirty` it was breaking in case we went from the details screen to filters and back. I prefer to stick with `isSavedAlertButtonDisabled`, in my opinion, this way it's clearer what we check